src/monster.cpp

```diff
@@ -162,6 +163,36 @@ void monster::poly(mtype *t)
     faction = t->default_faction;
 }
 
+void monster::update_check(){
+    if (type->upgrade_group == "NULL"){
+        return;
+    }
+    int current_day = calendar::turn.get_turn()/ DAYS(1);
+    int upgrade_time = type->upgrade_min * ACTIVE_WORLD_OPTIONS["MONSTER_GROUP_DIFFICULTY"];
+    //g->u.add_msg_if_player(m_debug, "Upgrade group: %s ", type->upgrade_group);
+    add_msg(m_debug, "Current:day: %d", current_day);
+    add_msg(m_debug, "Upgrade time : %d", upgrade_time);
+    add_msg(m_debug, "Last loaded: %d", last_loaded);
+
+    if (current_day == last_loaded || last_loaded < upgrade_time){
+        add_msg(m_debug, "Upgrade time less");
+        last_loaded = current_day;
+        return;
+    }
+
+    int time_passed = current_day - last_loaded;
+    add_msg(m_debug, "Time passed: %d", time_passed);
+    //radioactive decay function
+    float upgrade_chance =100 * (1- pow(0.5 , (time_passed / type->half_life ))) + type->base_upgrade_chance;
```

Comment:
time_passed and half_life are both ints and as such do integer division here as best as I can tell, which completely ruins what you are attempting to accomplish. Cast one to a float prior to the division or you are going to get some very unintended things happening.